addons/base/views.py

```diff
@@ -561,6 +676,64 @@ def addon_delete_file_node(self, target, user, event_type, payload):
                 file_node.delete(user=user)
 
 
+@file_signals.file_viewed.connect
+def osfstoragefile_mark_viewed(self, auth, fileversion):
+    if auth.user:
+        # mark fileversion as seen
+        FileVersionUserMetadata.objects.get_or_create(user=auth.user, file_version=fileversion)
+
+
+@file_signals.file_viewed.connect
+def osfstoragefile_update_view_analytics(self, auth, fileversion):
+    file = BaseFileNode.objects.get(versions__id=fileversion.id)
+    node = file.target
+    enqueue_update_analytics(
+        node,
+        file,
+        fileversion.identifier,
+        'view'
+    )
+
+
+@file_signals.file_viewed.connect
+def osfstoragefile_viewed_update_metrics(self, auth, fileversion):
+    resource = BaseFileNode.objects.get(versions__id=fileversion.id).target
```

Comment:
`FileVersion`s can be associated with more than one `BaseFileNode` -- this will raise a `MultipleObjectsReturned` in that case.